### ISSUE TITLE
Decompiler: reconstruct runtime async (MethodImplAttributes.Async) bodies

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#2742]
## Problem

The decompiler does not reconstruct **runtime async** methods (`MethodImplAttributes.Async`, no compiler-generated state machine). Only classic state-machine async is raised (`ClassicAsyncReconstructionPass`, which sets `RequiresAsyncBodyModifier`).

For runtime-async bodies:

- the `async` modifier is dropped (neither `ContainsAwaitExpression` nor `RequiresAsyncBodyModifier` is set), and
- the body exposes the raw runtime-async lowering (`AsyncHelpers.UnsafeAwaitAwaiter<...>`, `GetAwaiter`/`IsCompleted`/`GetResult`) instead of `await`.

### Verified repro (this SDK)

Composing an `async Task` / `async Task<int>` fixture built with SDK `11.0.100-preview.5` via `TypeSourceComposer.Compose(...)` renders, e.g.:

```csharp
public Task<int> AwaitingAsync()
{
    YieldAwaitable V_1 = Task.Yield();
    YieldAwaitable.YieldAwaiter V_0 = V_1.GetAwaiter();
    if (!V_0.IsCompleted)
    {
        AsyncHelpers.UnsafeAwaitAwaiter<YieldAwaitable.YieldAwaiter>(V_0);
    }
    V_0.GetResult();
    return 1;
}
```

Expected: `public async Task<int> AwaitingAsync() { await Task.Yield(); return 1; }`.

Note: the classic compiler state-machine async remains the language default; runtime async is a distinct emission shape (seen here from the preview SDK). The gap applies whenever runtime-async shapes appear.

## Secondary: async-signal inconsistency (classic async)

`TypeSourceComposer` forces `async` from `ContainsAwaitExpression` alone, while `ApiOutputFormatter` uses `ContainsAwaitExpression || RequiresAsyncBodyModifier`. For a classic-async method whose `await` is not reachable, the composer drops `async` where `ApiOutputFormatter` would not. Aligning the two is a small consistency fix (should ship with a classic-async fixture that exercises `RequiresAsyncBodyModifier` without `ContainsAwaitExpression`).

## Scope / discipline

Runtime-async recovery is a decompiler raise: follow `docs/decompiler-raise-discipline.md` (render-text A/B against merge-base, classify changed methods, adversarial review). Add both a real compiled runtime-async canary and near-miss negatives.

## Context

Surfaced while designing the member-body substrate (`docs/design/member-body-substrate.md`), filed as a followup so the substrate work stays scoped.

**Comment by richlander** [comment]
Confirmed on current `origin/main` with SDK `11.0.100-preview.7`: `PlainAsyncTask` containing `await Task.Yield()` renders the raw `UnsafeAwaitAwaiter` scaffold and drops `async`.

Two scope corrections before implementation:

1. Runtime async is already partially reconstructed. `AwaitRecoveryPass` raises exact corelib `AsyncHelpers.Await` calls for `Task`, `ValueTask`, and configured variants. This issue is specifically the additional **awaiter-helper/control-flow lowering**, not runtime async as a whole. A sharper title would be “Recover runtime-async awaiter-helper lowering.”
2. CoreLib exposes both `AsyncHelpers.AwaitAwaiter<TAwaiter>` and `AsyncHelpers.UnsafeAwaitAwaiter<TAwaiter>`. We should census compiler-produced uses of both before fixing only the demonstrated unsafe `Task.Yield` shape.

The strong gate should combine defining-method `MethodImplAttributes.Async` evidence with exact corelib helper identity/signature and same-awaiter-local correlation across `GetAwaiter`, `IsCompleted`, helper call, and `GetResult`. Today the defining method’s runtime-async fact does not reach the IR: `MethodRef.IsRuntimeAsync` exists for resolved callees, but `ImportedMethod`/`IrFunction` do not carry it. That is the first substrate addition; avoid a generic awaiter-pattern heuristic.

I would split the work:

- **Small consistency fix:** make `TypeSourceComposer` honor `ContainsAwaitExpression || RequiresAsyncBodyModifier`, matching `ApiOutputFormatter`, with the classic-async no-reachable-await fixture described here.
- **Decompiler raise:** recover the safe/unsafe awaiter-helper family, retaining the existing direct-`AsyncHelpers.Await` route. A dedicated runtime-async structural pass may be clearer than expanding the current direct-call rewrite, but the stepper trace should decide the insertion point.

The issue’s raise-discipline/evidence requirements are otherwise right. Positives should include compiled safe and unsafe awaiters plus sequential/control-flow cases; negatives should independently break method async metadata, corelib identity/signature, awaiter-local identity, `IsCompleted`, and `GetResult` correlation.
